modules/dnn/CMakeLists.txt

```diff
@@ -1,21 +1,17 @@
+set(the_description "Deep neural network module. It allows to load models from different frameworks and to make forward pass")
+ocv_add_module(dnn opencv_core opencv_imgproc WRAP python matlab java js)
+
 if(WINRT)
   ocv_module_disable(dnn)
```

Comment:
It is too late to disable module after "ocv_add_module()".
The same problem is below.

Reply:
@alalek, for a second case we can add
```cmake
if(DEFINED BUILD_PROTOBUF AND NOT BUILD_PROTOBUF)
  include(${OpenCV_SOURCE_DIR}/cmake/OpenCVFindLibProtobuf.cmake)
  if(NOT Protobuf_FOUND)
    ocv_module_disable(opencv_dnn)
  endif()
endif()
```
before `ocv_add_module`. Because if `BUILD_PROTOBUF` is `false` we'll try to link to system protobuf. If `BUILD_PROTOBUF` is `true` we can build it from 3rdparty anyway.

Reply:
What about `BUILD_${the_module}_INIT`?

Reply:
I believe `EXCLUDE_FROM_ALL` for `add_library()` + install(OPTIONAL) should help with protobuf,